xla/service/cpu/onednn_ops_rewriter.cc

```diff
@@ -61,11 +63,35 @@ HloInstruction* FindLayerNormShift(HloInstruction* instr) {
   return shift;
 }
 
-std::optional<HloInstruction*> MatchSoftmax(HloInstruction* instr) {
+bool is_neg_inf_const_scalar(const HloInstruction* const_instr) {
+  if (!ShapeUtil::IsEffectiveScalar(const_instr->shape())) return false;
+  auto value = LiteralUtil::GetFirstScalarLiteral(const_instr->literal());
+  return literal_comparison::Equal(
+             value, LiteralUtil::MinValue(const_instr->shape().element_type()))
+      .ok();
+}
+
+auto MaxReduce(HloInstruction** instr) {
+  auto is_valid_reduce_max = [](const HloInstruction* reduce) {
+    HloComputation* reducer = reduce->to_apply();
+    return (reducer->root_instruction()->opcode() == HloOpcode::kMaximum) &&
+           (reduce->dimensions().size() == 1) &&
+           (reduce->operand(1)->opcode() == HloOpcode::kConstant) &&
+           is_neg_inf_const_scalar(reduce->operand(1));
+  };
+
+  return m::AnyOf<HloInstruction>(
+      m::Maximum().WithBinaryOperandsAnyOrder(
```

Comment:
Is this particular match (`max(reduce, const)`) tested anywhere?

If not, please add a test for it.

Reply:
Yes, The test SoftmaxFP32OnAxisZero verifies this case.